core/src/main/java/org/apache/druid/timeline/SegmentWithOvershadowedStatus.java

```diff
@@ -21,16 +21,23 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
 
 /**
  * DataSegment object plus the overshadowed status for the segment. An immutable object.
  *
- * SegmentWithOvershadowedStatus's {@link #compareTo} method considers only the {@link SegmentId}
+ * DataSegmentWithOvershadowedStatus's {@link #compareTo} method considers only the {@link SegmentId}
  * of the DataSegment object.
  */
 public class SegmentWithOvershadowedStatus implements Comparable<SegmentWithOvershadowedStatus>
 {
   private final boolean overshadowed;
+  /**
+   * dataSegment is serialized "unwrapped", i.e. it's properties are included as properties of
+   * enclosing class. If in future, if {@Code SegmentWithOvershadowedStatus} were to extend {@link DataSegment},
```

Comment:
Suggest adding a test that verifies that the serialized form generated with the unwrapped annotation can be successfully deserialized as a subclass (can make a test impl) of DataSegment

Reply:
yeah, should test this, will try to add one.

Reply:
added a test